### PR TITLE
Tf composes

### LINKED ISSUE #1653: Improve getting of the Testing Farm composes
When getting a TF compose, we sanitize the target in [distro2compose method](https://github.com/packit/packit-service/blob/dad5a6d1c5eea1feb15a95652804cfc84083c0ca/packit_service/worker/helpers/testing_farm.py#L336). This breaks some usecases, e.g. having `RHEL-7.8-ZStream`  in the configuration results in `RHEL-7.8-Zstream` compose  (because of `title()`)  that is not in the list of available composes and therefore the test run fails ([example](https://github.com/r0x0d/leapp-packit-poc/pull/3/checks?check_run_id=8308160405)).

We could firstly check the raw distro (the target without arch) and if it is in the list of available composes, don't sanitize it. Or we could try to improve the sanitization itself, but we need to make sure we don't break the other usecases.

### PR DESCRIPTION
**Pull request by mfocko**
TODO:

- [X] Write new tests or update the old ones to cover new functionality.
- [X] Update doc-strings where appropriate.
- [X] Update or write new documentation in `packit/packit.dev`.

Fixes #1653

---

RELEASE NOTES BEGIN
We have improved mapping of Testing Farm Composes, if you have set your own custom mapping and the TF Compose is available, it will be used as is without any additional modifications we do (version, etc.).
RELEASE NOTES END

### CODE REVIEW

File: packit_service/worker/helpers/testing_farm.py [hunk 1]
```diff
@@ -354,6 +373,20 @@ def distro2compose(self, target: str) -> Optional[str]:
             compose if we were able to map the distro to compose present
             in the list of available composes, otherwise None
         """
+        composes = self.available_composes
+        if composes is None:
+            msg = "We were not able to get the available TF composes."
+            logger.error(msg)
+            self.report_status_to_test_for_test_target(
+                state=BaseCommitStatus.error,
```
Comment (lachmanfrantisek) [inline packit_service/worker/helpers/testing_farm.py:381]: :+1: Errors will make it possible to retrigger via the check-run button.

**Review by lachmanfrantisek — APPROVED** [review]
Thanks!

**Comment by lachmanfrantisek** [comment]
> TODO? do we need it in the release notes?

I think we can mention that if you pick anything from the list, we will not touch it, otherwise there is a mapping happening (= explain the behaviour and let users know that we've fixed a bug here).

**Review by lbarcziova — APPROVED** [review]
:+1:

**Comment by mfocko** [comment]
regate

**Pull request merged**